src/contexts/RestaurantListContext.jsx

```diff
@@ -0,0 +1,26 @@
+import { createContext, useState, useEffect } from "react";
+import { getRestaurantList } from "../api/restaurantAPI";
+
+export const RestaurantListContext = createContext();
+
+export function RestaurantListProvider({ children }) {
+  const [restaurantList, setRestaurantList] = useState([]);
+
+  const updateRestaurantList = () => {
+    getRestaurantList().then((data) => {
+      setRestaurantList(data);
+    });
+  };
```

Comment:
예외처리가 안되어있는 것 같아요! 또한 문법적으로 상관없지만 `getRestaurantList` 가 async함수로 작성되어있기 때문에 `.then`을 사용한 프로미스 체이닝보단 `await`를 쓰는게 더 적합해 보이는데 시간이 되신다면 리팩토링 부탁드립니다!
```suggestion
  const updateRestaurantList = async () => {
    try {
      const data = await getRestaurantList(); 
      setRestaurantList(data); 
    } catch (error) {
      console.error("Failed to fetch restaurant list:", error); 
    }
  };
```

Reply:
![image](https://github.com/user-attachments/assets/f76ddca5-9e8f-4e9a-9459-30cdc207d826)
예외처리는 step5 README에 "에러 상태는 고려하지 않습니다."라고 되어있어서 딱히 하지 않았었어요! 이번에 추가하도록 하겠습니다.
그리고 `.then`보다 `await`를 쓰는게 더 적합한 이유가 궁금합니다! 단순히 `async-await` 짝을 맞추기 위해서인가요? 아니면 다른 이유가 있나요?

Reply:
`async-await`을 쓰는 이유는 코드작성의 일관성 부분에서 말씀드린거긴합니다!